app/src/main/java/com/udacity/popularmovies/activity/MainActivity.java

```diff
@@ -246,19 +243,19 @@ public boolean onPrepareOptionsMenu(Menu menu) {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.action_most_popular_sort:
+                mSort = SortBy.MOST_POPULAR;
                 loadMovieData(SortBy.MOST_POPULAR);
                 item.setChecked(!item.isChecked());
-                mSort = SortBy.MOST_POPULAR;
                 return true;
             case R.id.action_top_rated_sort:
+                mSort = SortBy.TOP_RATED;
                 loadMovieData(SortBy.TOP_RATED);
                 item.setChecked(!item.isChecked());
-                mSort = SortBy.TOP_RATED;
```

Comment:
since the `loaderManager` now uses `mSort` to determine how to load the movie data, assignment needs to happen **before** the load is explicitly called on option menu click